### DISCUSSION TITLE
Publishing via github pages

### DISCUSSION THREAD

**Discussion by s3alfisc** [machow/quartodoc#322] [Q&A]
Hi all, 

I am currently trying to publish a quartodoc website for my package `PyFixest` via github pages, and it is a little bit unclear to me how to do it correctly. 

I am following the [quarto docs](https://quarto.org/docs/publishing/github-pages.html) and have rendered my `quartodoc` website locally to `/docs`, where everything looks super pretty =) 

Before pushing to github, I run `quarto render` & have linked github to the [development branch ](https://github.com/s3alfisc/pyfixest/pull/264)in which I move the docs to `quartodoc`. Unfortunately, [the site does not render properly](https://s3alfisc.github.io/pyfixest/benchmarks/readme.html). What could I be doing wrong? 

Here is my [development branch](https://github.com/s3alfisc/pyfixest/tree/quartodoc2). 

Any help would be greatly appreciated! 

All the best, Alex

**Comment by s3alfisc** [comment]
I actually think I found my mistake - I suppose I need to push `site_lib` to github?

**Reply by machow** [reply]
Hey, thanks for raising this. I took a look at the development branch, and think you're right about `site_libs`. 

I often put my `_quarto.yml` and all the doc pages (e.g. qmds, md files) into the `docs` folder. To render I do something like...

```
cd docs
quartodoc build --verbose
quarto render
```

Then, the rendered site ends up in `docs/_site`. I push everything in that to github pages.

The reason is that when `_quarto.yml` lives in the root, sometimes funky things happen (e.g. it fails to render something and leaves behind a folder or something).

Hope this helps!

**Reply by s3alfisc** [reply]
Thanks @machow! I have now put the `_quarto.yml` and everything else in `docs`. Do I still need to set `output-dir: docs` as explained in the [quarto documentation](https://quarto.org/docs/publishing/github-pages.html#render-to-docs)? Or should I point to `output-dir: _site`, as the [vetiver](https://github.com/rstudio/vetiver-python/blob/main/docs/_quarto.yml) docs do? I expect that I am still doing something wrong, as `gh-pages` rejects to render my `quartodoc`'s... 😅

**Reply by s3alfisc** [reply]
For now, I have just moved all content from `_site` into `docs`, and the [site is live and very pretty](https://s3alfisc.github.io/pyfixest/difference-in-differences.html)!
